src/managers/config.py

```diff
@@ -256,56 +271,107 @@ def scram_properties(self) -> list[str]:
+                f"listener.name.{name.lower()}.{mechanism.lower()}.sasl.jaas.config="
+                f"org.apache.kafka.common.security.scram.ScramLoginModule "
+                f'required username="{username}" password="{password}";'
+            )
+            scram_properties.append(
+                f"listener.name.{name.lower()}.sasl.enabled.mechanisms={mechanism}"
             )
 
         return scram_properties
 
     @property
-    def security_protocol(self) -> AuthMechanism:
-        """Infers current charm security.protocol based on current relations."""
-        # FIXME: When we have multiple auth_mechanims/listeners, remove this method
-        return (
-            "SASL_SSL"
-            if (self.state.cluster.tls_enabled and self.state.broker.certificate)
-            else "SASL_PLAINTEXT"
-        )
+    def oauth_properties(self) -> list[str]:
+        """Builds the properties for the oauth listener.
 
-    @property
-    def auth_mechanisms(self) -> list[AuthMechanism]:
-        """Return a list of enabled auth mechanisms."""
-        # TODO: At the moment only one mechanism for extra listeners. Will need to be
-        # extended with more depending on configuration settings.
-        protocol = [self.security_protocol]
-        if self.state.cluster.mtls_enabled:
-            protocol += ["SSL"]
+        Returns:
+            list of oauth properties to be set.
+        """
+        oauth_properties: list[str] = []
+        for listener in self.client_listeners:
+            if not listener.mechanism.startswith("OAUTH"):
+                continue
+            oauth_properties.append(
+                f"listener.name.{listener.name.lower()}.{listener.mechanism.lower()}.sasl.jaas.config="
+                "org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required \\\n"
+                '\toauth.client.id="kafka" \\\n'
+                f'\toauth.valid.issuer.uri="{self.oauth_config["issuer_url"]}" \\\n'
+                f'\toauth.jwks.endpoint.uri="{self.oauth_config["jwks_endpoint"]}" \\\n'
+                '\toauth.username.claim="email" \\\n'
+                '\toauth.check.audience="true" \\\n'
+                '\toauth.check.access.token.type="false" \\\n'
+                f'\toauth.config.id="{listener.name}" \\\n'
+                '\tunsecuredLoginStringClaim_sub="unused";'
+            )
+            oauth_properties.append(
+                f"listener.name.{listener.name.lower()}.{listener.mechanism.lower()}.sasl.server.callback.handler.class"
+                "=io.strimzi.kafka.oauth.server.JaasServerOauthValidatorCallbackHandler"
+            )
+            oauth_properties.append(
+                f"listener.name.{listener.name.lower()}.sasl.enabled.mechanisms="
+                f"{listener.mechanism}"
+            )
```

Comment:
**todo:** I don't love this string formatting, it's pretty obtuse with all the new-lines. I can see it tripping us up down the road. though I get where you were going with it being neater. 
IMO it's fine having a very long line in this case, as most editors wrap the text anyway.

Maybe do something like the following? Just to avoid all the new-line chars and multiple appends.

```python
long_oauth_properties = """
    prop.{var1}.{var2}='abc'
    other_prop.{var3}={var4}
"""

return [
    default_oauth_properties.format(var1='foo', var2='bar', var3='baz', var4='boo'),
]
```

Reply:
closing this since the formatting was updated and there are multiple comments on this sections